cheddar/src/internal.rs

```diff
@@ -105,27 +90,33 @@ impl Contract {
 
     /// Inner method to save the given account for a given account ID.
     /// If the account balance is 0, the account is deleted instead to release storage.
-    pub fn internal_update_account(&mut self, account_id: &AccountId, balance: u128) {
+    pub(crate) fn internal_update_account(&mut self, account_id: &AccountId, balance: u128) {
         if balance == 0 {
             self.accounts.remove(account_id);
         } else {
             self.accounts.insert(account_id, &balance); //insert_or_update
         }
     }
 
-    // TODO rename
-    pub fn int_ft_resolve_transfer(
+    /// Helper method to update balance of the sender and receiver based on the return
+    /// from the `on_ft_transfer` call.
+    /// Relper function which computes the amount refunded from the transfer_call and adjust
+    /// sender and receiver balances.
+    /// Returns: `(amount_credited_by_reciever, amount_burned)`, where
+    /// * amount_credited_by_receiver - is the amount transferred to the receiver after
+    ///   adjusting the balances
+    /// * amount_burned - when sender account is deleted we burn the unused tokens.
+    pub(crate) fn ft_resolve_transfer_adjust(
         &mut self,
         sender_id: &AccountId,
         receiver_id: ValidAccountId,
         amount: U128,
     ) -> (u128, u128) {
-        let sender_id: AccountId = sender_id.into();
         let receiver_id: AccountId = receiver_id.into();
         let amount: Balance = amount.into();
 
         // Get the unused amount from the `ft_on_transfer` call result.
-        let unused_amount = match env::promise_result(0) {
+        let mut unused_amount = match env::promise_result(0) {
```

Comment:
are you changing something?
it's not good to use a `mut` var when you can avoid it

Reply:
yes, later there is `refund_amount` which is basically "fixed" unused amount. It was easier to reason that we are adjusting the `unused_amount` based on the recipient balance and removing one more variable to remember.